x-pack/plugins/monitoring/server/alerts/disk_usage_alert.ts

```diff
@@ -0,0 +1,361 @@
+      });
+    } else {
+      const resolvedCount = alertStates.filter((alertState) => !alertState.ui.isFiring).length;
+      const resolvedNodes = (alertStates as AlertDiskUsageState[])
+        .filter((state) => !state.ui.isFiring)
+        .map((state) => `${state.nodeName}:${state.diskUsage.toFixed(2)}`)
+        .join(',');
+      if (resolvedCount > 0) {
+        const internalMessage = i18n.translate(
+          'xpack.monitoring.alerts.diskUsage.resolved.internalMessage',
+          {
+            defaultMessage: `Disk usage alert is resolved for {count} node(s) in cluster: {clusterName}.`,
+            values: {
+              count: resolvedCount,
+              clusterName: cluster.clusterName,
+            },
+          }
+        );
+
+        instance.scheduleActions('default', {
+          internalShortMessage: internalMessage,
+          internalFullMessage: internalMessage,
+          state: AlertingDefaults.ALERT_STATE.resolved,
+          nodes: resolvedNodes,
+          count: resolvedCount,
+          clusterName: cluster.clusterName,
+        });
+      }
+    }
+  }
+
+  protected processData(data: AlertData[], clusters: AlertCluster[], services: AlertServices) {
+    const currentUTC = +new Date();
+    for (const cluster of clusters) {
+      const nodes = data.filter((_item) => _item.clusterUuid === cluster.clusterUuid);
+      if (!nodes.length) {
+        continue;
+      }
+
+      const firingNodeUuids = nodes.reduce((list: string[], node) => {
+        const stat = node.meta;
+        if (node.shouldFire) {
+          list.push(stat.nodeId);
+        }
+        return list;
+      }, []);
+
+      firingNodeUuids.sort();
+
+      const instanceId = `${this.type}:${cluster.clusterUuid}:${firingNodeUuids.join(',')}`;
+      const instance = services.alertInstanceFactory(instanceId);
+      const state = instance.getState() as AlertInstanceState;
+      const alertInstanceState: AlertInstanceState = { alertStates: state?.alertStates || [] };
+
+      let shouldExecuteActions = false;
+      for (const node of nodes) {
+        const stat = node.meta as AlertDiskUsageState;
+
+        const indexInState = alertInstanceState.alertStates.findIndex((alertState) => {
+          const nodeAlertState = alertState as AlertDiskUsageState;
+          return (
+            nodeAlertState.cluster.clusterUuid === cluster.clusterUuid &&
+            nodeAlertState.nodeId === node.meta.nodeId
+          );
+        });
+        const nodeState = (indexInState > -1
+          ? alertInstanceState.alertStates[indexInState]
+          : this.getDefaultAlertState(cluster, node)) as AlertDiskUsageState;
+
+        nodeState.diskUsage = stat.diskUsage;
+        nodeState.nodeId = stat.nodeId;
+        nodeState.nodeName = stat.nodeName;
+        nodeState.ui.message = this.getUiMessage(nodeState, node);
+        shouldExecuteActions = shouldExecuteActions || node.shouldFire || nodeState.ui.isFiring;
```

Comment:
Yea so this is my fault, but this isn't really working as expected. The last part of this check ensures that we fire "resolution" actions if the state _was_ firing and is _not_ firing now. However, this most likely broke with [this code](https://github.com/elastic/kibana/pull/75419/files#diff-b2f11902f2e548816162b4da6341802bR313) where we create unique instance factories based on firing nodes uuids. 

I'm not sure how to properly solve this. [Official support](https://github.com/elastic/kibana/issues/49405) is most likely coming soon, but what we could do is maintain (in memory) a list of previously used `instanceIds` and then read the state from those and see if that tells us the scenario above (where it was firing but it's not now).

Thoughts?